assets/app/scripts/directives/nav.js

```diff
@@ -58,10 +58,11 @@ angular.module('openshiftConsole')
 
           sortedProjects = $filter('orderByDisplayName')(projects);
           options = _.map(sortedProjects, function(item) {
+            // truncates @ 35 as this prevents words from clobbering the caret
             return $('<option>')
-                      .attr("value", item.metadata.name)
+                      .attr('value', $filter('truncate')(item.metadata.name, 35))
```

Comment:
I can pull back on `value` and just truncate `text` I think.  I have a project made with a really long name and I see the text bleed out bad.

![screen shot 2016-02-04 at 4 24 14 pm](https://cloud.githubusercontent.com/assets/280512/12830143/c3edd6e6-cb5b-11e5-981b-0241be4ec794.png)